File: python/examples/user_orders.py
"""Fetch open orders for an authenticated user."""

import asyncio

from common import rest_client, get_keypair, login


async def main():
    client = rest_client()
    keypair = get_keypair()
    await login(client, keypair)
    pubkey = str(keypair.pubkey())
    print(f"logged in: {pubkey}")

    # 1. First page of user orders
    snapshot = await client.orders().get_user_orders(50)

    limit_orders = [o for o in snapshot.orders if o.order_type == "limit"]

    print(f"orders: {len(limit_orders)} limit")
    print(f"balances: {len(snapshot.balances)} market")
    print(f"has more: {snapshot.has_more}")

    if limit_orders:
        first = limit_orders[0]
        side = "BID" if first.side == 0 else "ASK"
        print(f"first limit: {first.order_hash} {side} @ {first.price}")

    # 2. Pagination
    if snapshot.next_cursor:
        page2 = await client.orders().get_user_orders(50, snapshot.next_cursor)
        print(f"next page: {len(page2.orders)} order(s)")
    else:
        print("no more pages")

    await client.close()


asyncio.run(main())

File: typescript/examples/user_orders.ts
import { restClient, getKeypair, login, runExample } from "./common";

async function main() {
  const client = restClient();
  const keypair = getKeypair();
  await login(client, keypair);

  const snapshot = await client.orders().getUserOrders(50);
  const limitOrders = snapshot.orders.filter(
    (order) => order.order_type === "limit"
  );

  console.log(`orders: ${limitOrders.length} limit`);
  console.log(`market balances: ${snapshot.market_balances.length}`);
  console.log(`has more: ${snapshot.has_more}`);

  const firstOrder = limitOrders[0];
  if (firstOrder) {
    console.log(
      `first limit: ${firstOrder.order_hash} ${firstOrder.side} @ ${firstOrder.price}`
    );
  }

  if (snapshot.has_more && snapshot.next_cursor) {
    const page2 = await client
      .orders()
      .getUserOrders(50, snapshot.next_cursor);
    console.log(`next page: ${page2.orders.length} order(s)`);
  }
}

void runExample(main);
